presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/PackageApiUtils.java

```diff
@@ -19,11 +19,14 @@
 import java.io.IOException;
 import java.io.InputStream;
 import java.net.URL;
+import java.nio.file.Path;
 
+import static com.google.common.io.MoreFiles.deleteDirectoryContents;
 import static com.teradata.prestomanager.agent.CommandExecutor.executeCommand;
 import static java.io.File.createTempFile;
 import static java.lang.String.format;
 import static java.nio.file.Files.copy;
+import static java.nio.file.Files.createDirectories;
 import static java.nio.file.StandardCopyOption.REPLACE_EXISTING;
 
 public class PackageApiUtils
```

Comment:
This class should be split into two classes, one for RPM, and one for tarballs.

The common method, `fetchFileFromURL` should be removed, and `AgentFileUtils::replaceFile` should be used instead (and renamed to indicate that it downloads a file).